From 5201dd1e92c825ba035a98557a7f49c491760464 Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Mon, 3 Feb 2020 18:36:32 +0800
Subject: [PATCH 1/2] fixed sensors name

---
 NodeManagerProxy.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/NodeManagerProxy.cpp b/NodeManagerProxy.cpp
index 54fa936..f45c7b4 100644
--- a/NodeManagerProxy.cpp
+++ b/NodeManagerProxy.cpp
@@ -121,13 +121,13 @@ void createSensors()
     // Global power statistics
     configuredSensors.push_back(std::make_unique<PowerMetric>(server));
     configuredSensors.push_back(std::make_unique<GlobalPowerPlatform>(
-        server, 0, 2040, "power", "Total_Power", globalPowerStats,
+        server, 0, 2040, "power", "total_power", globalPowerStats,
         entirePlatform, 0));
     configuredSensors.push_back(
-        std::make_unique<GlobalPowerCpu>(server, 0, 510, "power", "CPU_Power",
+        std::make_unique<GlobalPowerCpu>(server, 0, 510, "power", "cpu_power",
                                          globalPowerStats, cpuSubsystem, 0));
     configuredSensors.push_back(std::make_unique<GlobalPowerMemory>(
-        server, 0, 255, "power", "Memory_Power", globalPowerStats,
+        server, 0, 255, "power", "memory_power", globalPowerStats,
         memorySubsystem, 0));
 }
 
-- 
2.17.1

