From dd49a9c5e649a1eb383915ddb6fbed8da60838db Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Tue, 1 Jun 2021 11:44:18 +0800
Subject: [PATCH] driver: thermal: npcm_thermal: add hwmon sysfs

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 drivers/thermal/npcm_thermal.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/thermal/npcm_thermal.c b/drivers/thermal/npcm_thermal.c
index e88fe84c3f2b..c51910de1d8a 100644
--- a/drivers/thermal/npcm_thermal.c
+++ b/drivers/thermal/npcm_thermal.c
@@ -24,6 +24,7 @@
 #include <linux/regmap.h>
 #include <linux/mfd/syscon.h>
 #include "thermal_core.h"
+#include "thermal_hwmon.h"
 
 /* NPCM7xx GCR module */
 #define NPCM8XX_THRTL_OFFSET		0xC0
@@ -352,6 +353,8 @@ static int npcm_thermal_probe(struct platform_device *pdev)
 		if (irq > 0 && !priv->overheat_sensor)
 			npcm_configure_overheat_int(priv, tz, sensor->id);
 
+		if (devm_thermal_add_hwmon_sysfs(tz))
+			dev_warn(&pdev->dev, "failed to add hwmon sysfs attributes\n");
 	}
 
 	/* Just complain if no overheat interrupt was set up */
-- 
2.17.1

